### PR TITLE
Switch Tensorflow models tests back on Azure runners

### PR DESCRIPTION
**Pull request by ababushk**
### Details:
This switches TF Hub model tests back to self-hosted runners in AKS and stores models in persistent storage in Azure

### CODE REVIEW

**Comment by mryzhov** [comment]
looks like the memory issues have gone, remove temporary code and it is ready for merging

**Review by rkazants — CHANGES_REQUESTED** [review]
something with PyTorch models:

```
E             File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/huggingface_hub/file_download.py", line 1406, in hf_hub_download
E               raise LocalEntryNotFoundError(
E           
E           LocalEntryNotFoundError: An error happened while trying to locate the file on the Hub and we cannot find the requested files in the local cache. Please check your connection and try again or make sure your Internet connection is on.
```

@mvafin, please check this PR also

File: tests/model_hub_tests/tensorflow/test_tf_convert_model.py [hunk 1]
```diff
@@ -152,10 +152,10 @@ def clean_dir(self, dir_name: str):
 
     def teardown_method(self):
         # remove all downloaded files for TF Hub models
-        self.clean_dir(tf_hub_cache_dir)
+        # self.clean_dir(tf_hub_cache_dir)
 
         # remove all downloaded files for HF models
-        self.clean_dir(hf_cache_dir)
+        # self.clean_dir(hf_cache_dir)
```
Comment (rkazants) [inline tests/model_hub_tests/tensorflow/test_tf_convert_model.py:158]: please do not leave commented out code. If this is in production state, so let us remove it.
Do we cache both TF Hub and HF models?

Comment (ababushk) [reply tests/model_hub_tests/tensorflow/test_tf_convert_model.py:158]: I will remove them before the merge.
Yep, both directories were created on the shared drive. They reduce the execution time by an hour. It seems though that the cache doesn't work for pre-commit models scope.

File: .github/workflows/job_tensorflow_models_tests.yml [hunk 2]
```diff
@@ -71,6 +72,10 @@ jobs:
           echo "INSTALL_DIR=$GITHUB_WORKSPACE/install" >> "$GITHUB_ENV"
           echo "INSTALL_TEST_DIR=$GITHUB_WORKSPACE/install/tests" >> "$GITHUB_ENV"
           echo "MODEL_HUB_TESTS_INSTALL_DIR=$GITHUB_WORKSPACE/install/tests/model_hub_tests" >> "$GITHUB_ENV"
+          echo "TFHUB_CACHE_DIR=/mount/onnxtestdata$((GITHUB_RUN_NUMBER % NUMBER_OF_REPLICAS))/tfhub_models" >> "$GITHUB_ENV"
```
[the remark this answers is not in the document]
Comment (ababushk) [reply .github/workflows/job_tensorflow_models_tests.yml:75]: This is the historical name of our shared drive with test data

[the remark this answers is not in the document]
Comment (rkazants) [reply .github/workflows/job_tensorflow_models_tests.yml:75]: if it is easy to create a separate directory `tensorflow_models`, please create it and use.
Now it confuses and someone thinking of onnx can remove

[the remark this answers is not in the document]
Comment (ababushk) [reply .github/workflows/job_tensorflow_models_tests.yml:75]: I think we should rename it to just "testdata" and store all kinds of test data in subfolders

[the remark this answers is not in the document]
Comment (rkazants) [reply .github/workflows/job_tensorflow_models_tests.yml:75]: yes, please rename it and create separate sub-directory for each framework: `tensorflow`, `pytorch`, etc.

**Comment by rkazants** [comment]
@ababushk, what is speedup provided by caching? precommit wo caching vs. w caching? The same question for nightly scope.

File: .github/workflows/job_tensorflow_models_tests.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ababushk) [reply .github/workflows/job_tensorflow_models_tests.yml]: Currently it looks more like there are directories for each models source, e.g. `/mount/testdata0/hugging_face`, `/mount/testdata0/onnx_model_zoo`. `testdata1` is a full equivalent of `testdata0`, it was made to spread the IO load between two shared drives.

File: .github/workflows/job_tensorflow_models_tests.yml [hunk 3]
```diff
@@ -94,7 +99,7 @@ jobs:
         if: ${{ runner.os == 'Linux' }}
         run: |
           # install git (required to build pip deps from the sources)
-          sudo apt-get install --assume-yes --no-install-recommends g++ git ca-certificates
+          sudo apt-get install --assume-yes --no-install-recommends g++ git ca-certificates wget
```
Comment (rkazants) [inline .github/workflows/job_tensorflow_models_tests.yml:102]:
```suggestion
          sudo apt-get install --assume-yes --no-install-recommends g++ git ca-certificates
```

**Review by rkazants — APPROVED** [review]

**Pull request merged**